app/addons/auth/actions.js

```diff
@@ -0,0 +1,116 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+define([
+  'api',
+  'addons/auth/actiontypes'
+],
+function (FauxtonAPI, ActionTypes) {
+
+
+  var errorHandler = function (xhr, type, msg) {
+    msg = xhr;
+    if (arguments.length === 3) {
+      msg = xhr.responseJSON.reason;
+    }
+
+    FauxtonAPI.addNotification({
+      msg: msg,
+      type: 'error'
+    });
+  };
+
+
+  return {
+
+    login: function (username, password, session, urlBack) {
+      var promise = session.login(username, password);
+
+      promise.then(function () {
+        FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.loggedIn });
+        if (urlBack) {
+          return FauxtonAPI.navigate(urlBack);
+        }
+        FauxtonAPI.navigate('/');
+      });
+      promise.fail(errorHandler);
+    },
+
+    changePassword: function (password, passwordConfirm, session) {
+      var promise = session.changePassword(password, passwordConfirm);
+
+      promise.done(function () {
+        FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.changePassword });
+        FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CHANGE_PWD_FIELDS });
+      });
+
+      promise.fail(errorHandler);
+    },
+
+    updateChangePasswordField: function (value) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_FIELD,
+        value: value
+      });
+    },
+
+    updateChangePasswordConfirmField: function (value) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_CONFIRM_FIELD,
+        value: value
+      });
+    },
+
+    createAdmin: function (username, password, session, loginAfter) {
+      var promise = session.createAdmin(username, password, loginAfter);
+
+      promise.then(function () {
+        FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.adminCreated });
```

Comment:
You know, that's a fair point. I agree - I'll remove the explicit pass-by-props.